src/main/java/service/Validator.java

```diff
@@ -0,0 +1,38 @@
+package service;
+
+import java.util.function.Function;
+import model.FruitRecord;
+
+public class Validator implements Function<String, FruitRecord> {
+    private static final int ACTIVITY_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+
+    public FruitRecord apply(String rawRecord) {
```

Comment:
to follow SRP from SOLID it's better for Validator to be responsible only for validating rawRecord. Let's move creating of FruitRecord to parser 
```suggestion
    public void apply(String rawRecord) {
```